packages/client/src/comp/Address/Fields.tsx

```diff
@@ -0,0 +1,157 @@
+  const { fetchingData } = FetchingDataContainer.useContainer()
+  // When we first arrive at page, set focus and move cursor to beginning
+  useDidMount(() => {
+      if (type === 'initialAddress') {
+        if (path?.type === 'address' && addrRef?.current) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const controlEl = (addrRef.current as any).controlEl as HTMLInputElement
+          controlEl.focus({preventScroll: true})
+          controlEl.setSelectionRange(0, 0)
+        }
+      }
+  })
+
+  return <FlexBox>
+    {/* We use a 7:3 basis proportion between Flex Components */}
+    {/* The spacing between them 2% of the basis */}
+
+    <Flex basis='70%'>
+      <BaseInput
+        id='addr-street-input'  // This id is used for Warning Box to fill form quickly
+        label='Street Address'
+        ref={addrRef}
+        required
+        translate='no'
+        lang='en'
+        value={fields?.street}
+        onChange={e => setField('street', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='28%'>
+      <BaseInput
+        label='Apartment'
+        id='addr-apt-input'
+        translate='no'
+        lang='en'
+        value={fields?.unit}
+        onChange={e => setField('unit', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='40%'>
+      <BaseInput
+        id='addr-city-input'  // This id is used for Warning Box to fill form quickly
+        label='City'
+        required
+        translate='no'
+        lang='en'
+        value={fields?.city}
+        onChange={e => setField('city', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='28%' className='mui-select'>
+      <Select
+        id='addr-state-input'  // This id is used for Warning Box to fill form quickly
+        label='State'
+        translate='no'
+        lang='en'
+        onChange={e => {
+          // MuiCSS has a buggy support for <Select/> when using TypeScript,
+          // to really access HTMLSelect and its value we to do this hack
+          const trueSelect = e.currentTarget.firstChild as HTMLSelectElement
+          setField('state', trueSelect.value)
+        }}
+        defaultValue={fields?.state}
+        data-testid='addressFieldState'
+      >
+        {
+          [...allStates].sort().map((state) => {
+            return <Option
+              key={state}
+              value={state}
+              label={state}
+            >
+              {state}
+            </Option>
+          })
+        }
+      </Select>
+    </Flex>
+
+    <Flex basis='28%'>
+      <BaseInput
+        id='addr-zip-input'  // This id is used for Warning Box to fill form quickly
+        label='ZIP code'
+        required
+        translate='no'
+        lang='en'
+        pattern='[0-9]{5}'
+        value={fields?.postcode}
+        onChange={e => setField('postcode', e.currentTarget.value)}
+      />
+    </Flex>
+
+    {
+      type === 'initialAddress' && <Flex>
```

Comment:
Let's create a new `Address/Form.tsx` (`AddressForm.tsx` which includes `AddressInput` and this button).  No random if statements.